### PR TITLE
Disable model grad in occlusion sensitivity

### PR DESCRIPTION
**Pull request by Nic-Ma**
### Description
This PR added the missing logic to disable model grad for occlusion sensitivity metric.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by wyli** [comment]
I think this kind of logic should be done before passing the model into the metric computation. so not sure if it's needed here. @rijobro

**Comment by rijobro** [comment]
I initially put this in, but then took it back out again. If we use `model.eval()` then should we not return it to its original state at the end of the computation? We can do that fairly easily (`model.training`), but would add a bit of bloat so is perhaps unnecessary as @wyli says. I have no strong feelings either way.

**Comment by wyli** [comment]
thanks, I'm closing this for now

**Comment by Nic-Ma** [comment]
OK, let me remove model.eval().
But I think with torch.no_grad(): is necessary, right?
Even in training mode, it can make sure no effect to the grad computation and reduce memory usage.

Thanks.

**Comment by rijobro** [comment]
Removing `.eval` and keeping `no_grad` sounds good to me. Thanks!

**Comment by wyli** [comment]
I don't think no_grad is needed. in fact there's no need to assume `model` is an `nn.Module`. any callable would probably  work fine

**Comment by Nic-Ma** [comment]
Hi @rijobro @wyli ,

OK, thanks for the investigation, seems `detach()` already can disable the grad computation.
Let's close this PR for now.
Thanks for your review.
